src/routes/(pages)/blog/posts/nais-console/+page.md

```diff
@@ -0,0 +1,101 @@
+
+## The Technology Behind Nais Console
+
+The architecture of Nais Console is designed to scale and provide real-time insights into complex systems. At its core is the Nais API, which connects to all Kubernetes clusters and services like Prometheus. Initially, the system used an asynchronous approach for data collection and processing, but it has since transitioned to direct connections for more up-to-date and reliable information.
+
+For data collection and synchronization, we use Kubernetes Informers, which subscribe to specific changes in the cluster, such as pod events or resource updates. This event-driven model allows real-time updates in the Console without unnecessary system queries. Changes are captured and sent to the API, which uses Reconcilers to ensure the system’s actual state matches the desired state.
+
+Internally, we use Protocol Buffers (protobuf) for efficient data serialization, while externally, data is exposed via a GraphQL API. This allows clients to fetch only the data they need, avoiding unnecessary overhead. This setup provides flexibility and efficiency for both the frontend application and external integrations. In the future, we plan to open the API for broader use, including custom integrations and CLI tools.
+
+The Console’s frontend is built with Svelte, a modern and reactive JavaScript framework that balances high performance with low complexity. We also use Houdini, a GraphQL tool that simplifies caching and data management on the client side. Below is an example of how Houdini is used in the Nais Console frontend to fetch and display team information:
+
+```svelte
+<script lang="ts">
+  import { graphql } from '$houdini';
+  import { Skeleton } from '@nais/ds-svelte-community';
+
+  // Define a GraphQL query to fetch team information
+  const teamInfo = graphql(`
+    query TeamInfo($team: Slug!) @load {
+      team(slug: $team) @loading {
+        purpose @loading
+      }
+    }
+  `);
+</script>
+
+<div>
+  <!-- Display team purpose or a loading skeleton while data is being fetched -->
+  {#if $teamInfo.data}
+    {#if $teamInfo.data.team.purpose}
+      <p>Purpose: {$teamInfo.data.team.purpose}</p>
+    {:else}
+      <Skeleton variant="text" />
+    {/if}
+  {/if}
+</div>
+```
+
+This simplified example demonstrates how Svelte and Houdini work together to fetch and display data. The `graphql` function defines a query, and the `$teamInfo.data` store automatically updates when the query resolves. The `Skeleton` component provides a loading state while the data is being fetched.
+
+This example demonstrates how Houdini simplifies GraphQL queries and state management in the frontend. By leveraging Svelte’s reactivity and Houdini’s tools, the team can deliver a seamless and responsive user experience.
+
+This stack enables the team to build features quickly while maintaining a high degree of customization, ensuring that the Console remains a powerful and user-friendly tool for developers.
```

Comment:
```suggestion
This stack enables the Nais team to build features quickly while maintaining a high degree of customization, ensuring that the Console remains a powerful and user-friendly tool for developers.
```

Tredje paragrafen på rad som starter med `This`, blir bedre ved å fjerne forrige paragraf som diskutert ovenfor.